Sources/BSON/BSONError.swift

```diff
@@ -67,3 +67,68 @@ internal func BSONIterationError(
     error += " \(message)"
     return BSONError.InternalError(message: error)
 }
+
+/// Error thrown when a BSONValue type introduced by the driver (e.g. BSONObjectID) is encoded not using BSONEncoder
+internal func bsonEncodingUnsupportedError<T: BSONValue>(value: T, at codingPath: [CodingKey]) -> EncodingError {
+    let description = "Encoding \(T.self) BSONValue type with a non-BSONEncoder is currently unsupported"
```

Comment:
should we just say "BSON" instead of "BSONValue", since BSONValue is internal? (same applies to methods below)

Reply:
done